x-pack/plugins/apm/server/routes/suggestions/get_suggestions_with_term.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { rangeQuery } from '../../../../observability/server';
+import { Setup } from '../../lib/helpers/setup_request';
+
+export async function getSuggestionsWithTerm({
+  field,
+  filter,
+  setup,
+  size,
+  string,
+  start,
+  end,
+}: {
+  field: string;
+  filter: QueryDslQueryContainer[];
+  setup: Setup;
+  size: number;
+  string: string;
+  start?: number;
+  end?: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search('get_suggestions_with_terms', {
+    apm: {
+      events: [
+        ProcessorEvent.transaction,
+        ProcessorEvent.error,
+        ProcessorEvent.metric,
+      ],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...filter,
+            ...(start && end ? rangeQuery(start, end) : []),
+            {
+              query_string: {
+                fields: [field],
+                query: `*${string}*`,
+              },
+            },
+          ],
```

Comment:
Two things: 
 - what about using the `prefix` query? I think that's the behaviour of the terms_enum api that we are mimicking. 
 - Can we rename `field` to `fieldName` and `string` to `fieldValue`. I think we should consider doing the same for our suggestion APIs (I know that's what the terms enum API calls it but we can call it differently in our APIs).

```suggestion
          filter: [
            ...filter,
            prefix: { [fieldName]: fieldValue }
          ],
```

Reply:
I didn't know the existence of `prefix` query 🙈  but after experimenting a bit with it I believe it won't be as useful as the `query_string `.  

`prefix` returns the documents where the `fieldName` contains a term that **begins** with the `fieldValue`. I doubt the users will search with the termValue `https://` to get their environment which unfortunately are set with the host name.

Reply:
I tried `wildcard` query and it's much faster

Reply:
https://github.com/elastic/kibana/pull/126679/commits/ba53e9ea143169addfb522be69d551224522e50a